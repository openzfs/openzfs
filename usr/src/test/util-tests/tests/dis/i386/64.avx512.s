/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2017 Joyent, Inc.
 */

/*
 * Basic test for AVX512 mov instructions
 */

.text
.align 16
.globl libdis_test
.type libdis_test, @function
libdis_test:
	vmovaps		%xmm0, %xmm1
	vmovaps		%xmm2, %xmm3
	vmovaps		%xmm4, %xmm5
	vmovaps		%xmm6, %xmm7
	vmovaps		%xmm8, %xmm9
	vmovaps		%xmm10, %xmm11
	vmovaps		%xmm12, %xmm13
	vmovaps		%xmm14, %xmm15
	vmovaps		%xmm16, %xmm17
	vmovaps		%xmm18, %xmm19
	vmovaps		%xmm20, %xmm21
	vmovaps		%xmm22, %xmm23
	vmovaps		%xmm24, %xmm25
	vmovaps		%xmm26, %xmm27
	vmovaps		%xmm28, %xmm29
	vmovaps		%xmm30, %xmm31

	vmovaps		%ymm0, %ymm1
	vmovaps		%ymm2, %ymm3
	vmovaps		%ymm4, %ymm5
	vmovaps		%ymm6, %ymm7
	vmovaps		%ymm8, %ymm9
	vmovaps		%ymm10, %ymm11
	vmovaps		%ymm12, %ymm13
	vmovaps		%ymm14, %ymm15
	vmovaps		%ymm16, %ymm17
	vmovaps		%ymm18, %ymm19
	vmovaps		%ymm20, %ymm21
	vmovaps		%ymm22, %ymm23
	vmovaps		%ymm24, %ymm25
	vmovaps		%ymm26, %ymm27
	vmovaps		%ymm28, %ymm29
	vmovaps		%ymm30, %ymm31

	vmovaps		%zmm0, %zmm1
	vmovaps		%zmm2, %zmm3
	vmovaps		%zmm4, %zmm5
	vmovaps		%zmm6, %zmm7
	vmovaps		%zmm8, %zmm9
	vmovaps		%zmm10, %zmm11
	vmovaps		%zmm12, %zmm13
	vmovaps		%zmm14, %zmm15
	vmovaps		%zmm16, %zmm17
	vmovaps		%zmm18, %zmm19
	vmovaps		%zmm20, %zmm21
	vmovaps		%zmm22, %zmm23
	vmovaps		%zmm24, %zmm25
	vmovaps		%zmm26, %zmm27
	vmovaps		%zmm28, %zmm29
	vmovaps		%zmm30, %zmm31

	vmovaps		%zmm24, 0x8(%rsp)
	vmovaps		0x8(%rsp), %zmm31
	vmovaps		%zmm13, %zmm17{%k1}{z}
	vmovaps		%zmm16, %zmm24{%k7}
	vmovaps		%zmm28, %zmm29{z}

	vmovaps		%xmm16, %xmm25
	vmovaps		%xmm7, %xmm20{%k3}{z}
	vmovaps		%ymm1, %ymm16
	vmovaps		%ymm19, %ymm30{%k5}

	vmovapd		%zmm2, %zmm7
	vmovapd		%xmm16, %xmm25
	vmovapd		%ymm16, %ymm29

	vmovups		%zmm7, 0x20(%rbp)
	vmovups		0x8(%rsp), %zmm17
	vmovups		(%rax), %zmm23{z}
	vmovups		%xmm16, %xmm25
	vmovups		%ymm16, %ymm29

	vmovupd		(%r10), %zmm31{%k2}
	vmovupd		%zmm16, (%r11)
	vmovupd		%xmm16, %xmm25
	vmovupd		%ymm16, %ymm29

	/* Test offset handling for both disp8*N and full. */
	vmovaps		%zmm25, (%rsp)
	vmovaps		%zmm25, 0x20(%rsp)
	vmovaps		%zmm24, 0x40(%rsp)
	vmovaps		%zmm24, 0x60(%rsp)
	vmovaps		%zmm24, 0x80(%rsp)
	vmovaps		%zmm24, -0x80(%rsp)
	vmovaps		%zmm24, -0x20(%rsp)

	vmovaps		%ymm24, 0x10(%rsp)
	vmovaps		%ymm24, 0x20(%rsp)
	vmovaps		%ymm24, 0x40(%rsp)

	vmovaps		%xmm24, 0x8(%rsp)
	vmovaps		%xmm24, 0x10(%rsp)
	vmovaps		%xmm24, 0x20(%rsp)

	vmovaps		(%rsp), %zmm25
	vmovaps		0x20(%rsp), %zmm25
	vmovaps		0x40(%rsp), %zmm25
	vmovaps		0x60(%rsp), %zmm25
	vmovaps		0x80(%rsp), %zmm25
	vmovaps		-0x80(%rsp), %zmm25
	vmovaps		-0x20(%rsp), %zmm25

	vmovaps		0x10(%rsp), %ymm25
	vmovaps		0x20(%rsp), %ymm25
	vmovaps		0x40(%rsp), %ymm25

	vmovaps		0x8(%rsp), %xmm25
	vmovaps		0x10(%rsp), %xmm25
	vmovaps		0x20(%rsp), %xmm25

	vmovdqa32	%zmm6, 0x100(%rsp)
	vmovdqa32	%ymm26, 0x100(%rsp)
	vmovdqa32	%xmm16, 0x100(%rsp)
	vmovdqa32	(%rcx), %zmm6
	vmovdqa32	(%rcx), %ymm26
	vmovdqa32	(%rcx), %xmm16

	vmovdqa64	%zmm16, 0x100(%rsp)
	vmovdqa64	%ymm26, 0x100(%rsp)
	vmovdqa64	%xmm16, 0x100(%rsp)
	vmovdqa64	0x800(%rsp), %zmm16
	vmovdqa64	0x800(%rsp), %ymm26
	vmovdqa64	0x800(%rsp), %xmm16

	vmovdqu8	%zmm20, (%rsp)
	vmovdqu16	%zmm20, (%rsp)
	vmovdqu32	%zmm20, (%rsp)
	vmovdqu64	%zmm20, (%rsp)
	vmovdqu8	(%rsp), %zmm20
	vmovdqu16	(%rsp), %zmm20
	vmovdqu32	(%rsp), %zmm20
	vmovdqu64	(%rsp), %zmm20
.size libdis_test, [.-libdis_test]
