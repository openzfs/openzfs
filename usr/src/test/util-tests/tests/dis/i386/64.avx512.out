    libdis_test:       c5 f8 28 c8        vmovaps %xmm0,%xmm1
    libdis_test+0x4:   c5 f8 28 da        vmovaps %xmm2,%xmm3
    libdis_test+0x8:   c5 f8 28 ec        vmovaps %xmm4,%xmm5
    libdis_test+0xc:   c5 f8 28 fe        vmovaps %xmm6,%xmm7
    libdis_test+0x10:  c4 41 78 28 c8     vmovaps %xmm8,%xmm9
    libdis_test+0x15:  c4 41 78 28 da     vmovaps %xmm10,%xmm11
    libdis_test+0x1a:  c4 41 78 28 ec     vmovaps %xmm12,%xmm13
    libdis_test+0x1f:  c4 41 78 28 fe     vmovaps %xmm14,%xmm15
    libdis_test+0x24:  62 a1 7c 08 28 c8  vmovaps %xmm16,%xmm17
    libdis_test+0x2a:  62 a1 7c 08 28 da  vmovaps %xmm18,%xmm19
    libdis_test+0x30:  62 a1 7c 08 28 ec  vmovaps %xmm20,%xmm21
    libdis_test+0x36:  62 a1 7c 08 28 fe  vmovaps %xmm22,%xmm23
    libdis_test+0x3c:  62 01 7c 08 28 c8  vmovaps %xmm24,%xmm25
    libdis_test+0x42:  62 01 7c 08 28 da  vmovaps %xmm26,%xmm27
    libdis_test+0x48:  62 01 7c 08 28 ec  vmovaps %xmm28,%xmm29
    libdis_test+0x4e:  62 01 7c 08 28 fe  vmovaps %xmm30,%xmm31
    libdis_test+0x54:  c5 fc 28 c8        vmovaps %ymm0,%ymm1
    libdis_test+0x58:  c5 fc 28 da        vmovaps %ymm2,%ymm3
    libdis_test+0x5c:  c5 fc 28 ec        vmovaps %ymm4,%ymm5
    libdis_test+0x60:  c5 fc 28 fe        vmovaps %ymm6,%ymm7
    libdis_test+0x64:  c4 41 7c 28 c8     vmovaps %ymm8,%ymm9
    libdis_test+0x69:  c4 41 7c 28 da     vmovaps %ymm10,%ymm11
    libdis_test+0x6e:  c4 41 7c 28 ec     vmovaps %ymm12,%ymm13
    libdis_test+0x73:  c4 41 7c 28 fe     vmovaps %ymm14,%ymm15
    libdis_test+0x78:  62 a1 7c 28 28 c8  vmovaps %ymm16,%ymm17
    libdis_test+0x7e:  62 a1 7c 28 28 da  vmovaps %ymm18,%ymm19
    libdis_test+0x84:  62 a1 7c 28 28 ec  vmovaps %ymm20,%ymm21
    libdis_test+0x8a:  62 a1 7c 28 28 fe  vmovaps %ymm22,%ymm23
    libdis_test+0x90:  62 01 7c 28 28 c8  vmovaps %ymm24,%ymm25
    libdis_test+0x96:  62 01 7c 28 28 da  vmovaps %ymm26,%ymm27
    libdis_test+0x9c:  62 01 7c 28 28 ec  vmovaps %ymm28,%ymm29
    libdis_test+0xa2:  62 01 7c 28 28 fe  vmovaps %ymm30,%ymm31
    libdis_test+0xa8:  62 f1 7c 48 28 c8  vmovaps %zmm0,%zmm1
    libdis_test+0xae:  62 f1 7c 48 28 da  vmovaps %zmm2,%zmm3
    libdis_test+0xb4:  62 f1 7c 48 28 ec  vmovaps %zmm4,%zmm5
    libdis_test+0xba:  62 f1 7c 48 28 fe  vmovaps %zmm6,%zmm7
    libdis_test+0xc0:  62 51 7c 48 28 c8  vmovaps %zmm8,%zmm9
    libdis_test+0xc6:  62 51 7c 48 28 da  vmovaps %zmm10,%zmm11
    libdis_test+0xcc:  62 51 7c 48 28 ec  vmovaps %zmm12,%zmm13
    libdis_test+0xd2:  62 51 7c 48 28 fe  vmovaps %zmm14,%zmm15
    libdis_test+0xd8:  62 a1 7c 48 28 c8  vmovaps %zmm16,%zmm17
    libdis_test+0xde:  62 a1 7c 48 28 da  vmovaps %zmm18,%zmm19
    libdis_test+0xe4:  62 a1 7c 48 28 ec  vmovaps %zmm20,%zmm21
    libdis_test+0xea:  62 a1 7c 48 28 fe  vmovaps %zmm22,%zmm23
    libdis_test+0xf0:  62 01 7c 48 28 c8  vmovaps %zmm24,%zmm25
    libdis_test+0xf6:  62 01 7c 48 28 da  vmovaps %zmm26,%zmm27
    libdis_test+0xfc:  62 01 7c 48 28 ec  vmovaps %zmm28,%zmm29
    libdis_test+0x102: 62 01 7c 48 28 fe  vmovaps %zmm30,%zmm31
    libdis_test+0x108: 62 61 7c 48 29 84  vmovaps %zmm24,0x8(%rsp)
                       24 08 00 00 00 
    libdis_test+0x113: 62 61 7c 48 28 bc  vmovaps 0x8(%rsp),%zmm31
                       24 08 00 00 00 
    libdis_test+0x11e: 62 c1 7c c9 28 cd  vmovaps %zmm13,%zmm17{%k1}{z}
    libdis_test+0x124: 62 21 7c 4f 28 c0  vmovaps %zmm16,%zmm24{%k7}
    libdis_test+0x12a: 62 01 7c c8 28 ec  vmovaps %zmm28,%zmm29{z}
    libdis_test+0x130: 62 21 7c 08 28 c8  vmovaps %xmm16,%xmm25
    libdis_test+0x136: 62 e1 7c 8b 28 e7  vmovaps %xmm7,%xmm20{%k3}{z}
    libdis_test+0x13c: 62 e1 7c 28 28 c1  vmovaps %ymm1,%ymm16
    libdis_test+0x142: 62 21 7c 2d 28 f3  vmovaps %ymm19,%ymm30{%k5}
    libdis_test+0x148: 62 f1 fd 48 28 fa  vmovapd %zmm2,%zmm7
    libdis_test+0x14e: 62 21 fd 08 28 c8  vmovapd %xmm16,%xmm25
    libdis_test+0x154: 62 21 fd 28 28 e8  vmovapd %ymm16,%ymm29
    libdis_test+0x15a: 62 f1 7c 48 11 bd  vmovups %zmm7,0x20(%rbp)
                       20 00 00 00 
    libdis_test+0x164: 62 e1 7c 48 10 8c  vmovups 0x8(%rsp),%zmm17
                       24 08 00 00 00 
    libdis_test+0x16f: 62 e1 7c c8 10 38  vmovups (%rax),%zmm23{z}
    libdis_test+0x175: 62 21 7c 08 10 c8  vmovups %xmm16,%xmm25
    libdis_test+0x17b: 62 21 7c 28 10 e8  vmovups %ymm16,%ymm29
    libdis_test+0x181: 62 41 fd 4a 10 3a  vmovupd (%r10),%zmm31{%k2}
    libdis_test+0x187: 62 c1 fd 48 11 03  vmovupd %zmm16,(%r11)
    libdis_test+0x18d: 62 21 fd 08 10 c8  vmovupd %xmm16,%xmm25
    libdis_test+0x193: 62 21 fd 28 10 e8  vmovupd %ymm16,%ymm29
    libdis_test+0x199: 62 61 7c 48 29 0c  vmovaps %zmm25,(%rsp)
                       24 
    libdis_test+0x1a0: 62 61 7c 48 29 8c  vmovaps %zmm25,0x20(%rsp)
                       24 20 00 00 00 
    libdis_test+0x1ab: 62 61 7c 48 29 44  vmovaps %zmm24,0x40(%rsp)
                       24 01 
    libdis_test+0x1b3: 62 61 7c 48 29 84  vmovaps %zmm24,0x60(%rsp)
                       24 60 00 00 00 
    libdis_test+0x1be: 62 61 7c 48 29 44  vmovaps %zmm24,0x80(%rsp)
                       24 02 
    libdis_test+0x1c6: 62 61 7c 48 29 44  vmovaps %zmm24,-0x80(%rsp)
                       24 fe 
    libdis_test+0x1ce: 62 61 7c 48 29 84  vmovaps %zmm24,-0x20(%rsp)
                       24 e0 ff ff ff 
    libdis_test+0x1d9: 62 61 7c 28 29 84  vmovaps %ymm24,0x10(%rsp)
                       24 10 00 00 00 
    libdis_test+0x1e4: 62 61 7c 28 29 44  vmovaps %ymm24,0x20(%rsp)
                       24 01 
    libdis_test+0x1ec: 62 61 7c 28 29 44  vmovaps %ymm24,0x40(%rsp)
                       24 02 
    libdis_test+0x1f4: 62 61 7c 08 29 84  vmovaps %xmm24,0x8(%rsp)
                       24 08 00 00 00 
    libdis_test+0x1ff: 62 61 7c 08 29 44  vmovaps %xmm24,0x10(%rsp)
                       24 01 
    libdis_test+0x207: 62 61 7c 08 29 44  vmovaps %xmm24,0x20(%rsp)
                       24 02 
    libdis_test+0x20f: 62 61 7c 48 28 0c  vmovaps (%rsp),%zmm25
                       24 
    libdis_test+0x216: 62 61 7c 48 28 8c  vmovaps 0x20(%rsp),%zmm25
                       24 20 00 00 00 
    libdis_test+0x221: 62 61 7c 48 28 4c  vmovaps 0x40(%rsp),%zmm25
                       24 01 
    libdis_test+0x229: 62 61 7c 48 28 8c  vmovaps 0x60(%rsp),%zmm25
                       24 60 00 00 00 
    libdis_test+0x234: 62 61 7c 48 28 4c  vmovaps 0x80(%rsp),%zmm25
                       24 02 
    libdis_test+0x23c: 62 61 7c 48 28 4c  vmovaps -0x80(%rsp),%zmm25
                       24 fe 
    libdis_test+0x244: 62 61 7c 48 28 8c  vmovaps -0x20(%rsp),%zmm25
                       24 e0 ff ff ff 
    libdis_test+0x24f: 62 61 7c 28 28 8c  vmovaps 0x10(%rsp),%ymm25
                       24 10 00 00 00 
    libdis_test+0x25a: 62 61 7c 28 28 4c  vmovaps 0x20(%rsp),%ymm25
                       24 01 
    libdis_test+0x262: 62 61 7c 28 28 4c  vmovaps 0x40(%rsp),%ymm25
                       24 02 
    libdis_test+0x26a: 62 61 7c 08 28 8c  vmovaps 0x8(%rsp),%xmm25
                       24 08 00 00 00 
    libdis_test+0x275: 62 61 7c 08 28 4c  vmovaps 0x10(%rsp),%xmm25
                       24 01 
    libdis_test+0x27d: 62 61 7c 08 28 4c  vmovaps 0x20(%rsp),%xmm25
                       24 02 
    libdis_test+0x285: 62 f1 7d 48 7f 74  vmovdqa32 %zmm6,0x100(%rsp)
                       24 04 
    libdis_test+0x28d: 62 61 7d 28 7f 54  vmovdqa32 %ymm26,0x100(%rsp)
                       24 08 
    libdis_test+0x295: 62 e1 7d 08 7f 44  vmovdqa32 %xmm16,0x100(%rsp)
                       24 10 
    libdis_test+0x29d: 62 f1 7d 48 6f 31  vmovdqa32 (%rcx),%zmm6
    libdis_test+0x2a3: 62 61 7d 28 6f 11  vmovdqa32 (%rcx),%ymm26
    libdis_test+0x2a9: 62 e1 7d 08 6f 01  vmovdqa32 (%rcx),%xmm16
    libdis_test+0x2af: 62 e1 fd 48 7f 44  vmovdqa64 %zmm16,0x100(%rsp)
                       24 04 
    libdis_test+0x2b7: 62 61 fd 28 7f 54  vmovdqa64 %ymm26,0x100(%rsp)
                       24 08 
    libdis_test+0x2bf: 62 e1 fd 08 7f 44  vmovdqa64 %xmm16,0x100(%rsp)
                       24 10 
    libdis_test+0x2c7: 62 e1 fd 48 6f 44  vmovdqa64 0x800(%rsp),%zmm16
                       24 20 
    libdis_test+0x2cf: 62 61 fd 28 6f 54  vmovdqa64 0x800(%rsp),%ymm26
                       24 40 
    libdis_test+0x2d7: 62 e1 fd 08 6f 84  vmovdqa64 0x800(%rsp),%xmm16
                       24 00 08 00 00 
    libdis_test+0x2e2: 62 e1 7f 48 7f 24  vmovdqu8 %zmm20,(%rsp)
                       24 
    libdis_test+0x2e9: 62 e1 ff 48 7f 24  vmovdqu16 %zmm20,(%rsp)
                       24 
    libdis_test+0x2f0: 62 e1 7e 48 7f 24  vmovdqu32 %zmm20,(%rsp)
                       24 
    libdis_test+0x2f7: 62 e1 fe 48 7f 24  vmovdqu64 %zmm20,(%rsp)
                       24 
    libdis_test+0x2fe: 62 e1 7f 48 6f 24  vmovdqu8 (%rsp),%zmm20
                       24 
    libdis_test+0x305: 62 e1 ff 48 6f 24  vmovdqu16 (%rsp),%zmm20
                       24 
    libdis_test+0x30c: 62 e1 7e 48 6f 24  vmovdqu32 (%rsp),%zmm20
                       24 
    libdis_test+0x313: 62 e1 fe 48 6f 24  vmovdqu64 (%rsp),%zmm20
                       24 
